ubuntu-dev/ubuntu-dev.dockerfile

```diff
@@ -1,28 +1,29 @@
 FROM ubuntu:16.04
 
 # Install HTTPS transport for Ubuntu package sources.
-RUN apt-get update -q \
- && apt-get install -qy apt-transport-https
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends apt-transport-https ca-certificates software-properties-common \
+ && rm -rf /var/lib/apt/lists/*
 
 # Add source for the latest Clang packages.
 ADD llvm-snapshot.gpg.key /tmp
 RUN echo "deb https://apt.llvm.org/xenial/ llvm-toolchain-xenial-6.0 main" > /etc/apt/sources.list.d/llvm.list \
  && apt-key add /tmp/llvm-snapshot.gpg.key \
  && rm -f /tmp/llvm-snapshot.gpg.key
 
+# Add source for the latest Git packages.
+RUN add-apt-repository ppa:git-core/ppa
+
 # Add source for the latest Mercurial packages.
-RUN echo "deb http://ppa.launchpad.net/mercurial-ppa/releases/ubuntu xenial main" > /etc/apt/sources.list.d/mercurial.list \
- && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 41BD8711B1F0EC2B0D85B91CF59CE3A8323293EE
+RUN add-apt-repository ppa:mercurial-ppa/releases
```

Comment:
Note that this makes us a little more vulnerable to malicious key changes (e.g. if the PPA gets hacked).